### PR TITLE
KAFKA-5470: Replace -XX:+DisableExplicitGC with -XX:+ExplicitGCInvokesConcurrent in kafka-run-class

### PR DESCRIPTION
**Pull request by ijuma**
This is important because Bits.reserveMemory calls System.gc() hoping to free native
memory in order to avoid throwing an OutOfMemoryException. This call is currently
a no-op due to -XX:+DisableExplicitGC.

It's worth mentioning that -XX:MaxDirectMemorySize can be used to increase the
amount of native memory available for allocation of direct byte buffers.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5474/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5459/
Test PASSed (JDK 8 and Scala 2.12).

**Review by apurvam — APPROVED** [review]
LGTM. Seems like a reasonable default for the Kafka server, though we probably should call this out it in the upgrade notes of the next release.

**Comment by ijuma** [comment]
Review by @hachikuji.

**Comment by hachikuji** [comment]
@ijuma LGTM. I'll leave it up to you whether to merge to 0.11.0. If so, a mention in the release notes as Apurva suggests couldn't hurt.

**Comment by ijuma** [comment]
I'll merge this to trunk only and will wait until we release 0.11.0 to avoid merge conflicts in the upgrade notes.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5620/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5606/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by ijuma** [comment]
I added an upgrade note.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6191/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6207/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by omkreddy** [comment]
LGTM

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6376/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6361/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by ijuma** [comment]
@hachikuji, does the upgrade note look OK?

**Comment by hachikuji** [comment]
Yes, LGTM

**Pull request closed without merging**
